### ISSUE TITLE
mkdir Error on React built app's setup

### ISSUE THREAD

**Issue by Crevax** [electron-userland/electron-forge-templates#49]
* [x] I have read the [contribution documentation](https://github.com/electron-userland/electron-forge-templates/blob/master/CONTRIBUTING.md) for this project.
* [x] I agree to follow the [code of conduct](https://github.com/electron/electron/blob/master/CODE_OF_CONDUCT.md) that this project follows, as appropriate.
* [x] I have searched the issue tracker for an issue that matches the one I want to file, without success.

**Please describe your issue:**
Projects using the _react_ template attempt a mkdir on the `extensions` directory before the app's directory is created. Happening on OSX and Windows. Using the default ES6/7 template doesn't error. Below is a Windows screenshot.

![image](https://user-images.githubusercontent.com/431724/30559167-b732245e-9c81-11e7-842b-d51955fbaae0.png)

**Console output when you run `electron-forge` with the environment variable `DEBUG=electron-forge:*`. (Instructions on how to do so [here](https://www.npmjs.com/package/debug#usage)). Please include the stack trace if one exists.**

```
WARNING: DEBUG environment variable detected.  Progress indicators will be sent over electron-forge:lifecycle
  electron-forge:lifecycle Process Started: Checking your system +0ms
  electron-forge:lifecycle Process Succeeded: Checking your system +1s
  electron-forge:runtime-config setting key: verbose to value: false +0ms
WARNING: DEBUG environment variable detected.  Progress indicators will be sent over electron-forge:lifecycle
  electron-forge:lifecycle Process Started: Resolving Forge Config +0ms
  electron-forge:project-resolver searching for project in: C:\CQL_Work\CQL_Labs\electron-forge-playground +0ms
  electron-forge:project-resolver electron-forge compatible package.json found in C:\CQL_Work\CQL_Labs\electron-forge-playground\package.json +6ms
  electron-forge:lifecycle Process Succeeded: Resolving Forge Config +16ms
  electron-forge:require-search searching [ '../makers/win32/squirrel.js',
  '../makers/generic/squirrel.js',
  'electron-forge-maker-squirrel',
  'squirrel',
  'C:\\CQL_Work\\CQL_Labs\\electron-forge-playground\\squirrel',
  'C:\\CQL_Work\\CQL_Labs\\electron-forge-playground\\node_modules\\squirrel',
  'C:\\Users\\christopher.davis\\AppData\\Roaming\\nvm\\v8.5.0\\node_modules\\electron-forge\\dist\\makers\\win32\\squirrel.js',
  'C:\\Users\\christopher.davis\\AppData\\Roaming\\nvm\\v8.5.0\\node_modules\\electron-forge\\dist\\makers\\generic\\squirrel.js',
  'C:\\Users\\christopher.davis\\AppData\\Roaming\\nvm\\v8.5.0\\node_modules\\electron-forge\\dist\\api\\electron-forge-maker-squirrel',
  'C:\\Users\\christopher.davis\\AppData\\Roaming\\nvm\\v8.5.0\\node_modules\\electron-forge\\dist\\api\\squirrel',
  'C:\\CQL_Work\\CQL_Labs\\electron-forge-playground\\squirrel',
  'C:\\CQL_Work\\CQL_Labs\\electron-forge-playground\\node_modules\\squirrel',
  'C:\\Users\\christopher.davis\\AppData\\Roaming\\nvm\\v8.5.0\\node_modules\\electron-forge\\dist\\api\\makers\\win32\\squirrel.js',
  'C:\\Users\\christopher.davis\\AppData\\Roaming\\nvm\\v8.5.0\\node_modules\\electron-forge\\dist\\api\\makers\\generic\\squirrel.js',
  'C:\\Users\\christopher.davis\\AppData\\Roaming\\nvm\\v8.5.0\\node_modules\\electron-forge\\dist\\api\\node_modules\\electron-forge-maker-squirrel',
  'C:\\Users\\christopher.davis\\AppData\\Roaming\\nvm\\v8.5.0\\node_modules\\electron-forge\\dist\\api\\node_modules\\squirrel',
  'C:\\CQL_Work\\CQL_Labs\\electron-forge-playground\\squirrel',
  'C:\\CQL_Work\\CQL_Labs\\electron-forge-playground\\node_modules\\squirrel' ] relative to C:\Users\christopher.davis\AppData\Roaming\nvm\v8.5.0\node_modules\electron-forge\dist\api +0ms
  electron-forge:require-search testing ../makers/win32/squirrel.js +3ms
We need to package your application before we can make it
  electron-forge:lifecycle Process Started: Preparing to Package Application for arch: x64 +84ms
  electron-forge:project-resolver searching for project in: C:\CQL_Work\CQL_Labs\electron-forge-playground +92ms
  electron-forge:project-resolver electron-forge compatible package.json found in C:\CQL_Work\CQL_Labs\electron-forge-playground\package.json +1ms
  electron-forge:hook could not find hook: generateAssets +0ms
  electron-forge:hook could not find hook: prePackage +0ms
  electron-forge:packager packaging with options { asar: false,
  overwrite: true,
  packageManager: 'yarn',
  afterCopy: [ [Function], [Function], [Function], [Function] ],
  afterExtract: [],
  dir: 'C:\\CQL_Work\\CQL_Labs\\electron-forge-playground',
  arch: 'x64',
  platform: 'win32',
  out: 'C:\\CQL_Work\\CQL_Labs\\electron-forge-playground\\out',
  electronVersion: '1.7.6',
  quiet: true } +0ms
  electron-forge:lifecycle Process Succeeded: Preparing to Package Application for arch: x64 +19s
  electron-forge:lifecycle Process Started: Compiling Application +2ms
  electron-forge:lifecycle Process Succeeded: Compiling Application +2s
  electron-forge:lifecycle Process Started: Preparing native dependencies +1ms
  electron-forge:lifecycle Process Succeeded: Preparing native dependencies +2s
  electron-forge:lifecycle Process Started: Packaging Application +0ms
  electron-forge:hook could not find hook: postPackage +40s
  electron-forge:lifecycle Process Succeeded: Packaging Application +18s
Making for the following targets:
  electron-forge:hook could not find hook: preMake +2ms
  electron-forge:lifecycle Process Started: Making for target: squirrel - On platform: win32 - For arch: x64 +4ms
  electron-forge:lifecycle Process Succeeded: Making for target: squirrel - On platform: win32 - For arch: x64 +2m
  electron-forge:hook could not find hook: postMake +2m
```

**What command line arguments are you passing?**

```
none
```

**What does your `config.forge` data in `package.json` look like?**

```
"forge": {
    "make_targets": {
        "win32": [
            "squirrel"
        ],
        "darwin": [
            "zip"
        ],
        "linux": [
            "deb",
            "rpm"
        ]
    },
    "electronPackagerConfig": {
        "packageManager": "yarn"
    },
    "electronWinstallerConfig": {
        "name": "electron_forge_playground"
    },
    "electronInstallerDebian": {},
    "electronInstallerRedhat": {},
    "github_repository": {
        "owner": "",
        "name": ""
    },
    "windowsStoreConfig": {
        "packageName": "",
        "name": "electronforgeplayground"
    }
}
```

**Please provide either a failing minimal testcase (with a link to the code) or detailed steps to
reproduce your problem. Using `electron-forge init` is a good starting point, if that is not the
source of your problem.**

Just running `electron-forge init --template=react` and then `electron-forge make` was enough to trigger this error when running the built setup.exe.

**Comment by malept** [comment]
What happens when you try building a zip instead? I cannot repro this on Linux (with a zip).

**Comment by Crevax** [comment]
Looks like it's due to the installed version of electron-devtools-installer (see https://github.com/MarshallOfSound/electron-devtools-installer/commit/af4713414e81095161076d1ad67a9b3c6dc03f7f).

That fix was released in 2.1.0, so the caret version should have brought that in. But the init is setting an exact version of 2.0.1 for me. Not sure yet why that's happening (any pointers to help dig around?), but I'll open a PR to update the minimum dependencies since the older versions would break.

**Comment by Crevax** [comment]
After some more digging, the exact dependency version problem is a Windows-only issue, and it's due to the caret symbol having a special meaning in cmd (see https://github.com/yarnpkg/yarn/issues/3270).

Using double carets didn't work, but wrapping each dependency in double quotes gave the expected result running `yarn add` on the Windows command line (`npm install` worked fine without any modifications). It doesn't seem like any workaround is in progress in the yarn project. So would it make sense to open a new issue in the electron-forge project to add a special case when handling dependency setup on Windows?

**Comment by malept** [comment]
Yeah, that's really unfortunate.
